### ISSUE TITLE
Ability to pass to recipe positional arguments like $1, $2 or "$@"

### ISSUE THREAD

**Issue by alefbragin** [casey/just#803]
Currently the Just lacks ability to pass to recipe positional arguments like $1, $2 or "$@".

## Motivation

Example with proposed @-syntax for positional args:

```make
run_templ *cmd:
    docker run --rm alpine {{cmd}}

run_env *$cmd:
    docker run --rm alpine "${cmd}"

run_args *@cmd:
    docker run --rm alpine "$@"
```

Template w/o quotes :

```
$ just run_templ ls 'dir with space'
docker run --rm alpine ls dir with space
ls: dir: No such file or directory
ls: with: No such file or directory
ls: space: No such file or directory
```

Environment variable inside quotes :

```
$ just run_env ls 'dir with space'
docker run --rm alpine "${cmd}"
docker: Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: exec: "ls dir with space": executable file not found in $PATH: unknown.
```

Proper shell expansion of positional args:

```
$ just run_templ ls 'dir with space'
docker run --rm alpine "$@"
ls: dir with space: No such file or directory
```

Example with mixed arguments types:

```make
some_recipe named_template_arg @positional_arg_1 $JUST_ENV_VAR @positional_arg_2 another_named_template_arg *@positional_args_from_3_to_n:
    some_command {{named_template_arg}} {{another_named_template_arg}} "$@"
```

I don't sure about @-syntax. I only use it to demonstrate the idea of positional arguments.

**Comment by casey** [comment]
I decided that the best way to support this initially would be to add a justfile-wide setting that would cause positional arguments to be passed to all recipe commands:

```
set positional-arguments

foo bar:
  echo $1 # will print the value of `bar`
```

And just released v0.9.1 that includes this change.

This way, no specific syntax is required, and if the setting is turned on, you can switch between referring to arguments by position or interpolation as is convenient.

Try it out and let me know if it works for your use-case!

I'm actually considering making `set positional-arguments` the default in the future, probably with the `1.0` release. I opened an issue to track this here: #808.

**Comment by alefbragin** [comment]
> I'm actually considering making `set positional-arguments` the default in the future, probably with the `1.0` release.

I think it's a good decision!

> Try it out and let me know if it works for your use-case!

It works like a charm for my use-cases! Thanks!
